cancellation/activity.go

```diff
@@ -21,7 +21,7 @@ func (a *Activities) ActivityToBeCanceled(ctx context.Context) (string, error) {
 			activity.RecordHeartbeat(ctx, "")
 		case <-ctx.Done():
 			logger.Info("context is cancelled")
-			return "I am canceled by Done", nil
+			return "I am canceled by Done", ctx.Err()
```

Comment:
No reason to return a value if you're returning cancellation

Reply:
done :)